api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationCohortQueries.java

```diff
@@ -549,13 +457,13 @@ public CohortDefinition getPatientsWithRegimeDeTPT3HP() {
             + "                 AND e.encounter_type = ${60} "
             + "                 AND ( o.concept_id = ${23985} AND o.value_coded IN ( ${23954}, ${23984} ) ) "
             + "                 AND ( o2.concept_id = ${23987} AND o2.value_coded IN ( ${1256}, ${1705} ) ) "
-            + "                 AND e.encounter_datetime BETWEEN :startDate AND :endDate) filt "
+            + "                 AND e.encounter_datetime BETWEEN DATE_SUB(:endDate, INTERVAL 4 MONTH) AND :endDate ) filt "
             + "                 ON filt.patient_id = p.patient_id "
-            + "           WHERE p.voided = 0 AND e.voided = 0 AND o6.voided = 0 AND o7.voided = 0 "
+            + "           WHERE p.voided = 0 AND e.voided = 0 AND o6.voided = 0 "
             + "           AND e.location_id = :location "
             + "           AND e.encounter_type = ${53} "
             + "           AND o6.concept_id = ${23985} AND o6.value_coded = ${23954} "
-            + "           AND o7.concept_id = ${6128} AND o7.value_datetime <= Date_sub(filt.start_date, interval 4 month) "
```

Comment:
The 3HP start date must be computed according to new TPT structure. Check the requirements